browser/brave_cosmetic_resources_tab_helper.cc

```diff
@@ -0,0 +1,461 @@
+    for (size_t i = 0; i < force_hide_selectors_list->GetSize(); i++) {
+      if (i != 0) {
+        styled_stylesheet += ",";
+      }
+      styled_stylesheet += force_hide_selectors_list->GetList()[i].GetString();
+    }
+    styled_stylesheet += "{display:none!important;}\n";
+  }
+  base::DictionaryValue* style_selectors_dictionary = nullptr;
+  if (resources_dict->GetDictionary("style_selectors",
+      &style_selectors_dictionary)) {
+    for (const auto& it : style_selectors_dictionary->DictItems()) {
+      base::ListValue* style_selectors_list = nullptr;
+      if (!style_selectors_dictionary->GetList(it.first,
+          &style_selectors_list) || style_selectors_list->GetSize() == 0) {
+        continue;
+      }
+      styled_stylesheet += it.first + "{";
+      for (size_t i = 0; i < style_selectors_list->GetSize(); i++) {
+        if (i != 0) {
+          styled_stylesheet += ";";
+        }
+        styled_stylesheet += style_selectors_list->GetList()[i].GetString();
+      }
+      styled_stylesheet += ";}\n";
+    }
+    if (!styled_stylesheet.empty()) {
+      std::string cosmeticFilterConsiderNewSelectors_script =
+          "(function() {"
+            "let nextIndex = window.cosmeticStyleSheet.rules.length;";
+      cosmeticFilterConsiderNewSelectors_script +=
+            "window.cosmeticStyleSheet.insertRule(`" + styled_stylesheet +
+                "`, nextIndex);"
+            "window.allSelectorsToRules.set(`" +
+                styled_stylesheet + "`, nextIndex);"
+            "nextIndex++;";
+      cosmeticFilterConsiderNewSelectors_script +=
+            "if (!document.adoptedStyleSheets.includes("
+                  "window.cosmeticStyleSheet)){"
+               "document.adoptedStyleSheets = [window.cosmeticStyleSheet];"
+            "};";
+      cosmeticFilterConsiderNewSelectors_script +=
+          "})();";
+      auto* frame_host = content::RenderFrameHost::FromID(frame_id);
+      if (!frame_host)
+        return;
+      frame_host->ExecuteJavaScriptInIsolatedWorld(
+          base::UTF8ToUTF16(cosmeticFilterConsiderNewSelectors_script),
+          base::NullCallback(), ISOLATED_WORLD_ID_CHROME_INTERNAL);
+    }
+  }
+
+  if (!enabled_1st_party_cf_filtering_) {
+    auto* frame_host = content::RenderFrameHost::FromID(frame_id);
+    if (!frame_host)
+      return;
+    frame_host->ExecuteJavaScriptInIsolatedWorld(
+        base::UTF8ToUTF16(
+            *BraveCosmeticResourcesTabHelper::observing_script_),
+        base::NullCallback(), ISOLATED_WORLD_ID_CHROME_INTERNAL);
+  }
+}
+
+void BraveCosmeticResourcesTabHelper::GetHiddenClassIdSelectorsOnUI(
+    content::GlobalFrameRoutingId frame_id, const GURL& url,
+    std::unique_ptr<base::ListValue> selectors) {
+  if (!selectors || IsVettedSearchEngine(url.host())) {
+    return;
+  }
+  if (enabled_1st_party_cf_filtering_) {
+    // (sergz) We do the same in else section, we need to figure out do we
+    // need that if at all
```

Comment:
Not sure I understand the confusion here, but happy to clarify as necessary

Reply:
ok, I will contact you in DM to clarify that

Reply:
figured that out and make it right